sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -1917,9 +1918,12 @@ def func(df, *args, **kwargs):
       '__i%s__' % base,
       frame_base._elementwise_method('__i%s__' % base, inplace=True))
 
-for name in ['__lt__', '__le__', '__gt__', '__ge__', '__eq__', '__ne__']:
-  setattr(DeferredSeries, name, frame_base._elementwise_method(name))
-  setattr(DeferredDataFrame, name, frame_base._elementwise_method(name))
+for name in ['lt', 'le', 'gt', 'ge', 'eq', 'ne']:
+  for p in '%s', '__%s__':
+    # Note that non-underscore name is used for both as the __xxx__ methods are
+    # order-sensitive.
```

Comment:
That's odd. So I guess `series > other` will be one place where we diverge from the Pandas API (in an easily documented way).

Is there a similar issue for arithmetic operations?

Reply:
It appears just to be the comparison operators that suffer from this defect. 

For PCollections, the order is unspecified, so in some sense one can't say that comparison of "differently-ordered" dataframes should be rejected (as the notion of "differently-ordered" is not well defined).